sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -601,8 +601,9 @@ object DataSource extends Logging {
                 if (provider1.toLowerCase(Locale.ROOT) == "orc" ||
```

Comment:
As suggested in other PR, we can remove this condition.

Reply:
Yep. I'll remove this here since it's small.